src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string fileName = (data);
+        if (connFd < 0) {
+            instance_logger.log("Cannot accept connection to port " + to_string(serverPort), "error");
+        } else {
+            instance_logger.log("Connection successful to port " + to_string(serverPort), "info");
+            serviceArguments.connFd = connFd;
+            pthread_create(&threadA[connectionCounter], NULL, instanceservicesession, &serviceArguments);
+            // pthread_detach(threadA[connectionCounter]);
+            // pthread_join(threadA[connectionCounter], NULL);
+            connectionCounter++;
+        }
+    }
 
-            instance_logger.log("Received File name: " + fileName, "info");
-            write(connFd, JasmineGraphInstanceProtocol::SEND_FILE_LEN.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_FILE_LEN.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_LEN, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string size = (data);
-            instance_logger.log("Received file size in bytes: " + size, "info");
-            write(connFd, JasmineGraphInstanceProtocol::SEND_FILE_CONT.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_FILE_CONT.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_CONT, "info");
-            string fullFilePath =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + fileName;
+    for (int i = 0; i < connectionCounter; i++) {
+        pthread_join(threadA[i], NULL);
+        std::cout << "service Threads joined" << std::endl;
+    }
 
-            int fileSize = atoi(size.c_str());
-            while (true) {
-                if (Utils::fileExists(fullFilePath)) {
-                    while (Utils::getFileSize(fullFilePath) < fileSize) {
-                        bzero(data, INSTANCE_DATA_LENGTH + 1);
-                        read(connFd, data, INSTANCE_DATA_LENGTH);
-                        line = (data);
+    pthread_mutex_destroy(&file_lock);
+}
 
-                        if (line.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
-                            write(connFd, JasmineGraphInstanceProtocol::FILE_RECV_WAIT.c_str(),
-                                  JasmineGraphInstanceProtocol::FILE_RECV_WAIT.size());
-                        }
-                    }
-                    break;
-                } else {
-                    sleep(1);
-                    continue;
-                }
-            }
+int deleteGraphPartition(std::string graphID, std::string partitionID) {
+    int status = 0;
+    string partitionFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" +
```

Comment:
There are multiple calls for the same property value here. Each call to `getJasmineGraphProperty()` involves disk IO. Hence lets assign the value received from one call to `Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder")` a variable and use it throughout this function.

Reply:
Not each call need to read from disk. I have already fixed it in a previous PR. It's not loading the properties file at most once per worker (and once for master). It will keep it in memory and refer to the unordered_map for the future `getJasmineGraphProperty` calls.